TFT/src/User/API/SerialConnection.c

```diff
@@ -5,24 +5,72 @@
 #define SERIAL_PORT_2_QUEUE_SIZE 512
 #define SERIAL_PORT_3_QUEUE_SIZE 512
 #define SERIAL_PORT_4_QUEUE_SIZE 512
+#define SERIAL_PORT_5_QUEUE_SIZE 512
+#define SERIAL_PORT_6_QUEUE_SIZE 512
 
-SERIAL_PORT_INFO serialPort[SERIAL_PORT_COUNT] = {
-  {SERIAL_PORT, SERIAL_PORT_QUEUE_SIZE, false, "", "1 - Printer"},
+uint8_t serialActive = 0; // Bit array to store serial port state
+
+const SERIAL_PORT_INFO serialPort[SERIAL_PORT_COUNT] = {
+  {SERIAL_PORT, SERIAL_PORT_QUEUE_SIZE, "", "1 - Printer"},
+  #ifdef SERIAL_PORT_2
+    {SERIAL_PORT_2, SERIAL_PORT_2_QUEUE_SIZE, "2", "2 - WIFI"},
+  #endif
+  #ifdef SERIAL_PORT_3
+    {SERIAL_PORT_3, SERIAL_PORT_3_QUEUE_SIZE, "3", "3 - UART3"},
+  #endif
+  #ifdef SERIAL_PORT_4
+    {SERIAL_PORT_4, SERIAL_PORT_4_QUEUE_SIZE, "4", "4 - UART4"}
+  #endif
+  #ifdef SERIAL_PORT_5
+    {SERIAL_PORT_5, SERIAL_PORT_5_QUEUE_SIZE, "5", "5 - UART5"}
+  #endif
+  #ifdef SERIAL_PORT_6
+    {SERIAL_PORT_6, SERIAL_PORT_6_QUEUE_SIZE, "6", "6 - UART6"}
+  #endif
+};
+
+const uint8_t serialPortIndex[_UART_CNT] = {
```

Comment:
it's not correct because it depends on the TFT variants. For TFT28 index 0 could be UART_1 while for TFT35 index 0 could be UART_2 etc... For that reason I needed a not constant array and the value was detected and set in Serial_Init() with the instruction:
serialPortIndex[serialPort[i].port] = i;  // 1 for SERIAL_PORT_2 etc...

For backward compatibility I also maintained the "active" flag that for me it is useless once you have the possibility to disable the unused ports (so the check on active ports is not performed at all)

Reply:
solved